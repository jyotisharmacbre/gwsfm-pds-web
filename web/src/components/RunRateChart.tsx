import React from 'react';

import { Doughnut, Pie, Polar } from 'react-chartjs-2';
const data = {
    datasets: [{
      data: [
        11,
        16,
        32
        
      ],
      backgroundColor: [
        '#ced0fe',
        '#7e83fe','#704ba4'
       
      ],
      label: 'My dataset' // for legend
    }],
    labels: [
      'Opportunities',
      'J&A',
      'Lost'
    ]
  };
export default function RunRateChart(props: {Opportunities:number, JandA:number, Lost:number}){
    return (<div>
          <Pie  data={data} />
          </div>
             );
   
   
  }
  