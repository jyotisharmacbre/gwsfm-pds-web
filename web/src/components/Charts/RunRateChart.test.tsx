import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RunRateChart from './RunRateChart';
Enzyme.configure({ adapter: new Adapter() });


it('Run Rate Chart should render', () => {
    const component = shallow(
        <RunRateChart  ICE={56} JandA={12} BidSubmitted={30} OrderReceived={20} InProgress={24} Completed={20}/>
    );
    // let tree = component.toJSON();
    expect(component.getElements()).toMatchSnapshot();
});

 it('Graph Grid exist', () => {
   const props ={
    ICE:56, JandA:12, BidSubmitted:12, OrderReceived:34, InProgress:56, Completed:45
   }
     const component = mount(
        <RunRateChart {...props} />
    );

    expect(component.find('#GridRunRateChart').exists);  
 });


 
