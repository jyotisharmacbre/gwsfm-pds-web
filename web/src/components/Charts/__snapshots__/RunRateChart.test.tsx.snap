// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Run Rate Chart should render 1`] = `
Array [
  <ForwardRef(WithStyles)
    container={true}
    id="GridRunRateChart"
    spacing={0}
    style={
      Object {
        "height": "98%",
      }
    }
  >
     
    <ForwardRef(WithStyles)
      item={true}
      lg={6}
      md={6}
      sm={12}
      style={
        Object {
          "verticalAlign": "middle",
        }
      }
      xs={12}
    >
      <Polar
        data={
          Object {
            "datasets": Array [
              Object {
                "backgroundColor": Array [
                  "#CEF1A3",
                  "#526061",
                  "#FBD55B",
                  "#935FD6",
                  "#EA8032",
                  "#3DAAE9",
                ],
                "data": Array [
                  23,
                  16,
                  32,
                  12,
                  10,
                  12,
                ],
                "label": "My dataset",
              },
            ],
            "labels": Array [
              "Initial Customer Enquiry",
              "J&A",
              "Bid Submitted",
              "Order Received",
              "In Progress",
              "Completed",
            ],
          }
        }
        height={60}
        options={
          Object {
            "legend": Object {
              "display": true,
              "position": "bottom",
            },
            "maintainAspectRatio": true,
            "responsive": true,
          }
        }
        width={50}
      />
       
    </ForwardRef(WithStyles)>
     
    <ForwardRef(WithStyles)
      id="GridLostRunRateChart"
      item={true}
      lg={6}
      md={6}
      sm={12}
      style={
        Object {
          "textAlign": "center",
        }
      }
      xs={12}
    >
      <h4>
        Lost 
      </h4>
      <LostRunRateChart
        BidSubmitted={30}
        Completed={20}
        ICE={56}
        InProgress={24}
        JandA={12}
        OrderReceived={20}
      />
      <h4>
        On-Hold 
      </h4>
      <OnHoldRunRateChart
        BidSubmitted={30}
        Completed={20}
        ICE={56}
        InProgress={24}
        JandA={12}
        OrderReceived={20}
      />
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>,
]
`;
